lib/fluent/supervisor.rb

```diff
@@ -545,10 +545,10 @@ def dry_run_cmd
     end
 
     ## Set Engine's dry_run_mode true to override all target_id of worker sections
-    def dry_run
+    def dry_run(with_change_privilege = true)
       begin
         Fluent::Engine.dry_run_mode = true
-        change_privilege
```

Comment:
Supervisor process may have to change privilege to run `configure`. So it seems better that supervisor call `change_privilege` and worker process doesn't call `change_privilege` here:  https://github.com/fluent/fluentd/blob/5a0e14776b737ade4ca5a002bfbd0287d219dcb3/lib/fluent/supervisor.rb#L520

Reply:
In this situtaion, worker cann't change privilege when pass `--no-supervisor`, right?

Reply:
Ah, yes. Then it is good to call `change_privilege` on worker process when `--no-privilege` is specified and to call `change_privilege` on supervisor process when `--no-supervisor` isn't specified.